DeviceAdapters/WieneckeSinske/CAN29.cpp

```diff
@@ -23,12 +23,14 @@
 //
 //                IN NO EVENT SHALL THE COPYRIGHT OWNER OR
 //                CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-//                INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES.  
+//                INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES.
 //
 
 #ifdef WIN32
 #include <windows.h>
-#define snprintf _snprintf 
+#define snprintf _snprintf
+#elif __linux__
+#include <arpa/inet.h>
```

Comment:
Are these the only lines that are necessary to get the Linux build working? There are are a lot of diffs generated by (I assume) the addition/removal of spaces.

Reply:
Yes they are. Sorry for the formatting issue!

Reply:
Is there anything that I need to do beyond adding the #include statement to get the Linux build to work? I added the lines you suggest but I still run into the =error: ‘htonl’ was not declared in this scope= error. I checked and I have the inet header on my system, but maybe I need to set an environment variable before building?

Reply:
With that script https://github.com/hadim/mm_scripts/blob/master/build-mm.sh you should be able to install MM from a freshly installed Ubuntu.

Reply:
Thanks! The mistake was mine: I only added the include statement to one of two necessary .cpp files ;) Everything has compiled successfully now.